### PR TITLE
Add a callback to configure the overlaid view

### PR DESCRIPTION
**Pull request by TimothyChilvers**

### CODE REVIEW

File: Veneer/Classes/UIViewController+Veneer.swift [hunk 1]
```diff
@@ -27,7 +27,7 @@ extension DismissType: CustomStringConvertible {
 
 public extension UIViewController {
     
-    public func showVeneer(withHighlight highlight: Highlight, overlayViewType: VeneerOverlayView.Type = VeneerOverlayView.self) {
+    public func showVeneer(withHighlight highlight: Highlight, overlayViewType: VeneerOverlayView.Type = VeneerOverlayView.self, configurationCallback: ((VeneerOverlayView) -> ())? = nil) {
```
Comment (wattson12) [inline Veneer/Classes/UIViewController+Veneer.swift]: should the callback receive an instance of `overlayViewType`?

**Review by wattson12 — COMMENTED** [review]
Looks fine, just a question about the callback matching the overlayViewType in case it is subclassed

File: Veneer/Classes/UIViewController+Veneer.swift (hunk 1 above)
Comment (TimothyChilvers) [reply Veneer/Classes/UIViewController+Veneer.swift]: Yes, but you can't use a method argument in a different method parameter.

Comment (wattson12) [reply Veneer/Classes/UIViewController+Veneer.swift]: true would need to be generic i guess, but then would lose the default value e.g. 

```
public func showVeneer<T: VeneerOverlayView>(withHighlight highlight: Highlight, overlayViewType: T.Type = T.self, configurationCallback: ((T) -> ())? = nil)
```

We specify the overlay view type in every case at the moment anyway

**Pull request merged**